src/test/scala/org/bblf/client/BblfshClientTest.scala

```diff
@@ -128,5 +128,30 @@ class BblfshClientTest extends FunSuite with BeforeAndAfter {
     assert(endPos.get.line == 1)
     assert(endPos.get.col == 45)
   }
+
+  test("Test non increasing memory usage with filter()") {
+    System.gc()
+
+    def someFilter() = {
+      val node = resp.uast.get
+      client.filter(node, "//QualifiedName[@roleExpression]")
+      client.filter(node, "//*[@endLine='1']");
+      client.filter(node, "//*[@internalRole='types']");
+    }
+
+    def getMemUsage = {
+      System.gc()
+      Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()
+    }
+    someFilter
+    val initialUsage = getMemUsage
+
+    for (a<-1 to 10000) {
+      someFilter
+    }
+
+    val finalUsage = getMemUsage
+    assert(finalUsage <= initialUsage * 1.03)
```

Comment:
This seems pretty ad-hoc... and will probably make this test unstable.
Using relative values also doesn't help.

Reply:
Yeah, it's imprecise which unittests shouldn't be; but it runs in the context of the unittests which is somewhat of a controlled environment. I'll remove it anyway.